src/compat.h

```diff
@@ -676,4 +676,90 @@ pg_sub_s64_overflow(int64 a, int64 b, int64 *result)
 }
 #endif
 
+/* Backport of list_qsort() */
+#if PG11_LT
+typedef int (*list_qsort_comparator)(const void *a, const void *b);
```

Comment:
Not sure if a `List` is necessary. Isn't it easier to just allocate a chunk of memory and store pointers to jobs there and use normal `qsort`?

That means less code to maintain, less code that can contain bugs, and easier memory management.

Reply:
I think the function `ts_update_scheduled_jobs_list` actually works a lot better if the jobs are in a list. Or are you suggesting converting from list to array only for sorting in the scheduler? If we do decide to convert the jobs list to an array, I'd suggest doing it in a separate PR.

Reply:
I was suggesting to use an array instead of a list, yes. Ok for doing the change in a separate PR (might need to be discussed as well, so better to not let it block anything.)